tests/drivers/can/api/src/canfd.c

```diff
@@ -37,6 +37,34 @@ static void tx_std_callback_2(const struct device *dev, int error, void *user_da
 	zassert_equal(frame->id, TEST_CAN_STD_ID_2, "ID does not match");
 }
 
+static void tx_std_callback_netdown_1(const struct device *dev, int error, void *user_data)
+{
+	const struct can_frame *frame = user_data;
+
+	if ((error == 0) || (error == -ENETDOWN)) {
+		/* Frame may be sent before CAN is stopped */
+		k_sem_give(&tx_callback_sem);
+	}
+
+	zassert_equal(dev, can_dev, "CAN device does not match");
+	zassert_equal(frame->id, TEST_CAN_STD_ID_1, "ID does not match");
+	TC_PRINT("error1: %d\n", error);
+}
+
+static void tx_std_callback_netdown_2(const struct device *dev, int error, void *user_data)
+{
+	const struct can_frame *frame = user_data;
+
+	if ((error == 0) || (error == -ENETDOWN)) {
+		/* Frame may be sent before CAN is stopped */
+		k_sem_give(&tx_callback_sem);
+	}
+
+	zassert_equal(dev, can_dev, "CAN device does not match");
+	zassert_equal(frame->id, TEST_CAN_STD_ID_2, "ID does not match");
+	TC_PRINT("error2: %d\n", error);
+}
```

Comment:
These are not CAN FD specific. Please test this in `classic.c` only.

Reply:
This shall check following conditional statement (condition meet):
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcan.c#L364
Moved to classic.c.